src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,42 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] checkDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            checkDiagonal[i] = matrix[i][i];
+        }
+        return checkDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] checkCounterDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            checkCounterDiagonal[i] = matrix[i][matrix.length - i - 1];
+        }
+        return checkCounterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = Integer.MAX_VALUE;
```

Comment:
can you solve without min and max integer values?

Reply:
done